### PR TITLE
Fix : go lint, go fmt code and add contribution docs

### PR DESCRIPTION
**Pull request by aananthraj**
This commit

	- lints code using `gofmt -w -s .` to lint code
	- adds CONTRIBUTING.md and updates README.md
	- adds go report card badge and removes broken slack badge in README.md.
	- adds comments to Exported functions and variables in packages

Thoughts and Comments ?

### CODE REVIEW

**Comment by aananthraj** [comment]
@jbianquetti-nami @jjo, 

This PR is to enhance code lint and fmt. I have added go report card badge instead of broken slack badge. Kindly review this and let me know changes/modifications if any.

**Comment by aananthraj** [comment]
Looping @arapulido. Can you help here ?

**Comment by jjo** [comment]
> Looping @arapulido. Can you help here ?

Hi @aananthraj, thanks much for this PR !

Also note that we work on different timezones, helping on maintaining this
project as time and priorities permit. As you've seen in previous PR we did
put time on doing thorough code reviews, as we'll do with this one.

**Comment by aananthraj** [comment]
Yes @jjo, I can understand.
> Also note that we work on different timezones, helping on maintaining this
project as time and priorities permit.

I do appreciate the teams efforts. Having a common forum would help to avoid communication lags. Sorry for being in hurry.

**Comment by aananthraj** [comment]
ping (a gentle reminder)

**Comment by aananthraj** [comment]
@jjo,  @jbianquetti-nami any updates on this PR.
can this be merged..?

**Comment by tylerauerbeck** [comment]
@aananthraj I took a look at this previously and from a quick glance everything looks good (from the listing, formatting perspective).

As for the contribution perspective -- I'll hold any comments and wait for @jbianquetti-nami and @jjo to weigh in here.

**Comment by aananthraj** [comment]
@tylerauerbeck thanks for the reiview.
Hope, we could merge this soon ☺️.

File: config/config.go [hunk 1]
```diff
@@ -149,6 +151,7 @@ func (c *Config) Load() error {
 	return nil
 }
 
+// CheckMissingResourceEnvvars checks the missing environment variables
```
Comment (jjo) [inline config/config.go]: Suggest rewording as: 
```
// CheckMissingResourceEnvvars will read the environment for equivalent config variables to set
```

File: docs/CONTRIBUTION.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Contributing to Kubewatch
+
+We love your input! We want to make contributing to this project as easy and transparent as possible, whether it's:
+
+- Reporting a bug
+- Discussing the current state of the code
+- Submitting a fix
+- Proposing new features
+
+## We Develop with Github
```
Comment (jjo) [inline docs/CONTRIBUTION.md]: Suggest rewording as:
```
## We use Github to develop kubewatch
```

File: docs/CONTRIBUTION.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+We use github to host code, to track issues and feature requests, as well as accept pull requests.
+
+## We Use [Github Flow](https://guides.github.com/introduction/flow/index.html), So All Code Changes Happen Through Pull Requests
```
Comment (jjo) [inline docs/CONTRIBUTION.md]: Please un-CamelCase the 2nd part of this sentence, maybe leaving `Pull Requests` only as so.

File: pkg/controller/controller.go [hunk 4]
```diff
@@ -65,6 +65,7 @@ type Controller struct {
 	eventHandler handlers.Handler
 }
 
+// Start kubewatch server
```
Comment (jjo) [inline pkg/controller/controller.go]: Suggest rewording:
```
// Start prepares watchers and run their controllers, then waits for process termination signals
```

**Comment by aananthraj** [comment]
@jjo thanks for approving this PR. The requested changes were added :+1: .

**Comment by aananthraj** [comment]
Thanks everyone, 👍
